File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;
using System.Linq;
using System.Collections.Generic;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        var decompose = number;
        var length = 0;

        while (decompose > 0)
        {
            decompose = decompose / 10;
            length++;
        }

        var sum = 0d;
        decompose = number;

        while (decompose > 0)
        {
            var digit = decompose % 10;
            sum += Math.Pow(digit, length);
            decompose = decompose / 10;
        }

        return number == sum;
    }
}

File: fsharp/armstrong-numbers/ArmstrongNumbers.fs
module ArmstrongNumbers

let rec digits(number: int): int list =
  if number = 0 then []
  elif number < 10 then [number]
  else
    let digit = number % 10
    let outNumber = (number - digit) / 10
    let outDigits = digits(outNumber)
    outDigits @ [digit]

let rec sum(allDigits: int list, length: int) =
  match allDigits with
    | head::tail -> (pown head length) + sum(tail, length)
    | _ -> 0

let isArmstrongNumber (number: int): bool =
  let allDigits = digits number
  let allDigitsLength = List.length allDigits
  let totalSum = sum(allDigits, allDigitsLength)
  number = totalSum

